src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1951,6 +1956,202 @@ def _add_virtual_node_role_assignment(cmd, result, vnet_subnet_id):
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterAgentPoolProfile = self.cmd.get_models(
+            "ManagedClusterAgentPoolProfile",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedClusterIdentity = self.cmd.get_models(
+            "ManagedClusterIdentity",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        # pylint: disable=line-too-long
+        self.ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties = self.cmd.get_models(
+            "ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ManagedCluster = self.cmd.get_models(
+            "ManagedCluster",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        # pylint: disable=line-too-long
+        self.Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties = self.cmd.get_models(
+            "Components1Umhcm8SchemasManagedclusteridentityPropertiesUserassignedidentitiesAdditionalproperties",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ExtendedLocation = self.cmd.get_models(
+            "ExtendedLocation",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+        self.ExtendedLocationTypes = self.cmd.get_models(
+            "ExtendedLocationTypes",
+            resource_type=self.resource_type,
+            operation_group="managed_clusters",
+        )
+
+
+# pylint: disable=too-few-public-methods
+class AKSCreateParameters:
+    # used to store original function parameters, in the form of attributes of this class, which can be
+    # obtained or set through a.xxx (a is an instance of this class, xxx is the original parameter name)
+    # Note: the attributes of this class should not be modified once they are initialized
+    def __init__(self, data):
+        for name, value in data.items():
+            setattr(self, name, value)
+
+
+# pylint: disable=too-few-public-methods
+class AKSCreateContext:
+    # used to store intermediate variables and parameters that are not specified by the user but are automatically
+    # completed/patched (neither original function parameters nor models)
+    def __init__(self):
+        pass
+
+
+class AKSCreateDecorator:
+    def __init__(self, cmd, client, models, raw_parameters, resource_type=ResourceType.MGMT_CONTAINERSERVICE):
+        self.cmd = cmd
+        self.client = client
+        self.models = models
+        self.param = AKSCreateParameters(raw_parameters)
+        self.context = AKSCreateContext()
```

Comment:
could you please add some comments describing which kind of data should be stored to the context.
so people can follow.